sdk/analytics/src/test/java/com/klaviyo/analytics/state/KlaviyoStateTest.kt

```diff
@@ -281,4 +285,88 @@ internal class KlaviyoStateTest : BaseTest() {
         assertEquals(state.email, null)
         assertEquals(state.phoneNumber, null)
     }
+
+    @Test
+    fun `createEvent buffers event when no observers are registered`() {
+        mockkObject(TrackedEventsBuffer)
+        try {
+            val event = Event(EventMetric.CUSTOM("test_event"))
+            val profile = Profile()
+
+            state.createEvent(event, profile)
+
+            verify { TrackedEventsBuffer.addEvent(event) }
+        } finally {
+            unmockkObject(TrackedEventsBuffer)
+        }
+    }
+
+    @Test
+    fun `createEvent invokes observer when observer is registered`() {
+        var invokedEvent: Event? = null
+        val observer: ProfileEventObserver = { event -> invokedEvent = event }
+
+        state.onProfileEvent(observer)
+
+        val testEvent = Event(EventMetric.CUSTOM("test_event"))
+        val profile = Profile()
+
+        state.createEvent(testEvent, profile)
+
+        assertEquals(testEvent, invokedEvent)
+    }
+
+    @Test
+    fun `createEvent does not buffer when observer is registered`() {
+        mockkObject(TrackedEventsBuffer)
+        try {
+            val observer: ProfileEventObserver = { _ -> }
+            state.onProfileEvent(observer)
+
+            val event = Event(EventMetric.CUSTOM("test_event"))
+            val profile = Profile()
+
+            state.createEvent(event, profile)
+
+            verify(exactly = 0) { TrackedEventsBuffer.addEvent(any()) }
+        } finally {
+            unmockkObject(TrackedEventsBuffer)
+        }
+    }
+
+    @Test
+    fun `createEvent invokes multiple observers when registered`() {
+        var invokedCount = 0
+        val observer1: ProfileEventObserver = { _ -> invokedCount++ }
+        val observer2: ProfileEventObserver = { _ -> invokedCount++ }
+
+        state.onProfileEvent(observer1)
+        state.onProfileEvent(observer2)
+
+        val event = Event(EventMetric.CUSTOM("test_event"))
+        val profile = Profile()
+
+        state.createEvent(event, profile)
+
+        assertEquals(2, invokedCount)
+    }
+
+    @Test
+    fun `createEvent buffers again after all observers are removed`() {
```

Comment:
Reading through tests can be very elucidating... Some more mild concerns that I don't have answers to yet
1. What happens if the host app never calls `registerForInAppForms` -- I think we just keep doing our 10s buffer forever.
2. If they wait a little while to register for in app forms, could we wind up injecting events you wouldn't really expect to trigger a form.
3. Similarly, if they unregister for a period, then re-register, again we might wind up injecting events in an unexpected way there.

Reply:
I don't have answers yet, just trying to think through scenarios